functions/src/utils/agora.js

```diff
@@ -0,0 +1,41 @@
+const functions = require("firebase-functions");
+
+const { RtcTokenBuilder, RtcRole } = require("agora-access-token");
+
+const AGORA_CONFIG = functions.config().agora;
+
+const appId = AGORA_CONFIG.app_id;
+const appCertificate = AGORA_CONFIG.app_certificate;
+
+// @debt should we move this into AGORA_CONFIG, or maybe venue config in firestore or similar?
+const expirationTimeInSeconds = 3600;
+
+const getExpirationTime = () => {
+  const currentTimestamp = Math.floor(Date.now() / 1000);
+
+  return currentTimestamp + expirationTimeInSeconds;
+};
+
+const generateAgoraTokenForAccount = ({ channelName, account, role }) => {
+  // @debt we should enforce a stricter security requirement on channelName. Maybe use UUIDs?
+  if (typeof channelName !== "string" && channelName.length <= 0)
+    throw new Error("channelName must be a string");
+
+  if (typeof account !== "string") throw new Error("account must be a string");
```

Comment:
[NB] just to be consistent we could also check if `account` is not empty ( as we do for `channelName` few lines above)
```suggestion
  if (typeof account !== "string" && account.length <= 0)
    throw new Error("account must be a string");
```

Reply:
Sounds like a good idea :)

Reply:
fixed in https://github.com/sparkletown/sparkle/pull/1673/commits/7176004c307d345508fb782d7713f4f0684cc672